src/apps/search-result/search-result.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from "react";
+import SearchResultList from "../../components/search-result-list/search-result.list";
+import SearchResultPager from "../../components/search-result-pager/search-result-pager";
+import {
+  SearchWithPaginationQuery,
+  useSearchWithPaginationQuery
+} from "../../core/dbc-gateway/generated/graphql";
+import { usePrevious } from "../../core/utils/helpers";
+
+interface SearchResultProps {
+  q: string;
+  numberOfResultItems: number;
+}
+
+const SearchResult: React.FC<SearchResultProps> = ({
+  q,
+  numberOfResultItems
+}) => {
+  const [works, setWorks] = useState<
+    SearchWithPaginationQuery["search"]["works"] | []
+  >([]);
+  const [hitcount, setHitCount] = useState<
+    SearchWithPaginationQuery["search"]["hitcount"] | number
+  >(0);
+  const [page, setPage] = useState(1);
+  const [searchItemsShown, setSearchItemsShown] = useState(numberOfResultItems);
+
+  const setPageHandler = () => {
+    const currentPage = page + 1;
+    setPage(currentPage);
+    setSearchItemsShown(currentPage * numberOfResultItems);
+  };
+
+  const { isSuccess, data } = useSearchWithPaginationQuery({
+    q: { all: q },
+    offset: page === 1 ? 0 : searchItemsShown,
+    limit: numberOfResultItems
+  });
+
+  const firstWork = data?.search?.works?.[0]?.id;
```

Comment:
maybe a comment to explain these consts?

Reply:
Yes! Good point.